### PR TITLE
[MINOR] Fluent API for Batch

### PR DESCRIPTION
**Pull request by vinniefranco**
This is the start of an `Ecto.Multi` style API for `Exandra.Batch`

### CODE REVIEW

File: lib/exandra/connection.ex [hunk 1]
```diff
@@ -98,6 +98,17 @@ defmodule Exandra.Connection do
 
   @impl Ecto.Adapters.SQL.Connection
   def query(cluster, sql, params, opts) do
+    receiver = Keyword.get(opts, :send_raw, false)
+
+    if receiver do
+      send(receiver, {:raw, sql, params})
```
Comment (vinniefranco) [inline lib/exandra/connection.ex:104]: I am not sure how I feel about this approach, but, Ecto doesn't really provide an easy way to get the equivalent of `to_sql` for `:insert`, `:update`, or `:delete`

File: lib/exandra/batch.ex [hunk 2]
```diff
@@ -23,8 +23,137 @@ defmodule Exandra.Batch do
...
+       Exandra.Batch.new()
+       |> Exandra.Batch.insert(:admin, User.changeset(%{email: "jeff@example.com"})
+       |> Exandra.Batch.insert(:consumer, User.changeset(%{email: "britta@example.com"})
+
+      Exandra.execute_batch(MyApp.Repo, batch)
   """
 
+  @type t :: %__MODULE__{
+          status:
+            :default
```
Comment (vinniefranco) [inline lib/exandra/batch.ex:38]: Supports legacy

File: lib/exandra/connection.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/exandra/connection.ex:104]: @vinniefranco yeah agreed that this feels weird. Ecto calls back to the `insert` function in `Exandra.Connection` for this, maybe there's a way we can use that? By converting the changeset into a bunch of fields first, basically. 🤔

Comment (vinniefranco) [reply lib/exandra/connection.ex:104]: I went with that approach initially. What I didn't like was the amount of duping from Repo.Schema.do_insert that would be a lot of surface area to maintain and keep in sync. 

Ideally .to_sql gets updated to work with insert and friends.

Comment (whatyouhide) [reply lib/exandra/connection.ex:104]: Yeah, fair enough. I can ask around and understand why Ecto doesn't provide that already.

**Pull request closed without merging**